### PR TITLE
Add ability to set a default work type per-tenant

### PR DESCRIPTION
**Pull request by bkiahstroud**
Ref:
- https://github.com/scientist-softserv/hykuup_knapsack/issues/168

### CODE REVIEW

File: app/controllers/admin/work_types_controller.rb [hunk 1]
```diff
@@ -12,9 +12,9 @@ def edit
       site
     end
 
+    # TODO: extend to handle :default_work_type
```
Comment (bkiahstroud) [inline app/controllers/admin/work_types_controller.rb:15]: We can remove this now

File: app/models/site.rb [hunk 2]
```diff
@@ -81,4 +83,10 @@ def remove_admins_by_email(emails)
       u.remove_role :admin, self
     end
   end
+
+  def default_work_type_is_available
+    return true if available_works.include?(default_work_type)
+
+    errors.add(:default_work_type, "#{default_work_type} is not enabled in this tenant. Available work types are: #{available_works.join(', ')}")
```
Comment (bkiahstroud) [inline app/models/site.rb:90]: Thanks for fixing this, I couldn't remember the syntax off the top of my head 😅

File: app/views/admin/work_types/edit.html.erb [hunk 3]
```diff
@@ -5,14 +5,31 @@
...
-      <% Hyrax.config.registered_curation_concern_types.each do |type| %>
-        <div class="form-check">
-          <input class="form-check-input" type="checkbox" value="<%= type %>" id="input-<%= type %>" name="available_works[]" <%= 'checked' if @site.available_works&.include?(type) %>>
-          <label class="form-check-label" for="input-<%= type %>">
-            <%= type %>
-          </label>
-        </div>
-      <% end %>
+      <fieldset>
+	<legend>Choose the available work types:</legend>
```
Comment (bkiahstroud) [inline app/views/admin/work_types/edit.html.erb:9]: Should we add this text to `i18n`?

File: app/views/admin/work_types/edit.html.erb [hunk 4]
```diff
@@ -5,14 +5,31 @@
...
+            <input class="form-check-input" type="checkbox" value="<%= type %>" id="input-<%= type %>" name="site[available_works][]" <%= 'checked' if @site.available_works&.include?(type) %>>
+            <label class="form-check-label" for="input-<%= type %>">
+              <%= type %>
+            </label>
+	  </div>
+	<% end %>
+      </fieldset>
+
+      <fieldset>
+	<legend>Choose the default work type for this site:</legend>
```
Comment (bkiahstroud) [inline app/views/admin/work_types/edit.html.erb:21]: Should we add this text to `i18n`?

**Comment by kirkkwang** [comment]
currently seeing this error, when you re-evaluate the line that's erroring in the web console, it seems to be fine
image

**Comment by bkiahstroud** [comment]
@kirkkwang @ShanaLMoore do either of you know what the latest for this PR is? Did this get solved for Mobius in a different way? [The ticket](https://github.com/scientist-softserv/hykuup_knapsack/issues/168) is closed, but I don't see another related PR

**Comment by ShanaLMoore** [comment]
I'm not sure, I've added LaRita to your tag since she and Kirk were the ones who worked on Mobius.

**Comment by laritakr** [comment]
I think the decision was just to not do it, as far as I remember. It wasn't working correctly, so we just set the available work types on the tenant's config.

**Pull request closed without merging**